### DISCUSSION TITLE
git commit問題（抱歉老師我不是故意的）

### DISCUSSION THREAD

**Discussion by Yunhans** [1101-GS3073-C/Python-Community-GS3073-C#40] [Q&A]
我在測試git push和git commit的時候搞了很久，心情不太愉快，輸入了一些dirty words，搞了半天改不回來，請問該如何更改commit呢？
請老師大發慈悲原諒我，我會認真學習AI的
<img width="1440" alt="截圖 2021-10-09 下午4 03 38" src="https://user-images.githubusercontent.com/91072133/136651060-b605ce3f-948f-4337-a826-c9cb0411d06c.png">

**Comment by jacky0115** [comment]
0. 開啟終端機
1. 輸入`git log --oneline`可以查看commit的歷史紀錄  
2. 輸入`git rebase -i HEAD~5`代表我要修改最近5次的commit，可以依據你要修改的commit調整`5`這個數字
3. 此時終端機會跳出 pick \<commit id> \<commit message>，像我這邊輸入5，所以出現5行的 pick \<commit id> \<commit message>
4. 輸入`i`進入vim的insert模式，把`pick`修改成 `r` 或 `reword`，按`esc`離開insert模式，再輸入`:wq`儲存離開
5. 接著他會依照你剛剛修改的reword分次跳出vim編輯器，此時一樣輸入`i`進入vim的insert模式，就可以換成你想要的commit message了，換完之後，按`esc`離開insert模式，再輸入`:wq`儲存離開，你reword改多少次，這些動作就會重複多少次，全部改完後應該會看到successfully rebased and updated
6. 最後重新push到github的repository 可以輸入`git push origin main`，或用vscode的GUI來push，如果出現錯誤訊息，改成強制push，輸入`git push origin main -f`

或是你覺得以上步驟很麻煩的話，如果這個repository不重要，也可以直接刪掉這個Repository重新開一個，在檔案總管中你當初使用`git init`的資料夾裡面會有一個.git檔(如果沒看到的話要點`檢視`->勾選`隱藏的項目`才看的到)，把它刪掉就可以消除整個git的歷史紀錄，但他也會因此變成普通的資料夾

**Reply by Yunhans** [reply]
非常感謝

**Reply by enoch-lightning** [reply]
@jacky0115 我可以問你是怎麼知道要照上面這些步驟操作的嗎？是你試試看試出來的結論，還是網路上查到的資料？如果是網路上查到的資料，能否請你分享？謝謝你。
